internal/cli/kraft/cloud/compose/build/build.go

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+package build
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/spf13/cobra"
+
+	kraftcloud "sdk.kraft.cloud"
+	kcinstances "sdk.kraft.cloud/instances"
+
+	"kraftkit.sh/cmdfactory"
+	"kraftkit.sh/compose"
+	"kraftkit.sh/config"
+	"kraftkit.sh/internal/cli/kraft/cloud/utils"
+	"kraftkit.sh/internal/cli/kraft/pkg"
+	"kraftkit.sh/log"
+	"kraftkit.sh/packmanager"
+	"kraftkit.sh/unikraft/app"
+)
+
+type BuildOptions struct {
+	Auth   *config.AuthConfig           `noattribute:"true"`
+	Client kcinstances.InstancesService `noattribute:"true"`
+	Metro  string                       `noattribute:"true"`
+	Token  string                       `noattribute:"true"`
+
+	project     *compose.Project
+	composeFile string
```

Comment:
(Comment applies to all commands)

Since those aren't attributes which can set directly by the user via command-line flags, do they need to be inside this struct at all?

I'm excluding `metro` and `token`, which are set in `Pre` and therefore need to be carried to `Run`.
https://github.com/unikraft/kraftkit/blob/1f3487e1c216deb6e966a67da33baaadca2d4c91/internal/cli/kraft/cloud/tunnel/tunnel.go#L26-L29

Reply:
hmm, makes sense, I just added them there because I saw them similar to `metro/token` + it made copy-paste easier

I'll move them down as variables

Reply:
fixed as new commit